test/should_pass/opaque.erl

```diff
@@ -0,0 +1,13 @@
+-module(opaque).
+
+-export([external/2, internal/2]).
+
+-spec external(user_types:my_opaque(), integer() | undefined) -> integer().
+external(_, undefined) -> 0;
+external(_, I) -> I.
+
+-opaque my_opaque() :: integer().
+
+-spec internal(my_opaque(), integer() | undefined) -> integer().
+internal(_, undefined) -> 0;
```

Comment:
I'd expect a test that the type is still visible from inside the module, i.e. that the first parameter is used in this function.